pages/en/network/developing.mdx

```diff
@@ -2,36 +2,39 @@
+Subgraph developers often curate their own subgraph. This proves to Indexers that they've published a high-quality subgraph to the network and attracts additional Curators.
 
 Curators stake GRT with the subgraph and receive curation shares, which guarantee a percentage of future query fees. The curation signal increases as more Curators signal on the subgraph.
 
-A strong curation signal is crucial to a subgraph. Indexers use the curation signal to determine which subgraphs to index to the network, making it available for querying.
+A strong curation signal is crucial because Indexers use the curation signal to determine which subgraphs to index to the network, making it available for querying.
 
-### Deploying on the network
+### Publishing on the Network
 
-By deploying your subgraph to The Graph network, you're making that data available to developers and users across the web3 ecosystem. Anyone will be able to query it.
+By deploying your subgraph to The Graph Network, you're making that data available to developers and users across the web3 ecosystem. Anyone will be able to query it.
 
 Data from your subgraph can be used to improve user experience, build new features, or even create entirely new applications.
 
 ## Querying Subgraphs
 
-Developers that query subgraphs serve it to users through a simple interface. They can use the GraphQL API to query data directly from the blockchain. Developers can manipulate, sort, or aggregate the data any way they want.
+Developers that query subgraphs serve it to users through a simple interface. They can use the GraphQL API to query data from the subgraph's processed database. Developers can manipulate, sort, or aggregate the data any way they want.
+
+When you query a subgraph on The Graph Network, it will be served by one of the Indexers who has chosen to index that subgraph. Queries incur query fees, which are paid in GRT. These fees are managed and deducted via a Gateway, which matches consumers and Indexers, and keeps track of consumer spending via API keys.
 
-Querying a subgraph with your API key will generate query fees that can be paid in GRT.
 
 ### How Billing Works
 
+> IMPORTANT: When an account reaches a 0 GRT balance and the billing invoice is not paid in a timely manner, Indexers will stop serving queries for all API keys associated with the account. Indexers will automatically start serving queries again once the billing balance is paid in full and GRT is added to the account for future query fees.
+
 Developers that query a subgraph can use GRT to pay the query fees. This fee is split between different network participants, including the Indexers, Curators, and Delegators.
 
 Your account on [Subgraph Studio](https://thegraph.com/studio/) must have a positive balance of GRT, which will be used to pay for query fees to the network. Maintaining a positive balance for an account is the responsibility of the user or dapp.
```

Comment:
I think an additional line is required here to explain what is happening with the Studio + the Gateway here

"Query payments are facilitated by Gateways, which match incoming queries with suitable indexers. Consumers must use an API key when making queries on the network via a Gateway. API keys can be managed via the Subgraph Studio, including keeping API keys sufficiently funded with GRT to pay for queries. Indexers receive vouchers for answering queries, which can be redeemed for GRT from the API key's balance.

Reply:
Got that line in there!